docs/userguide/format_parse/numbers/measurement_units.md

```diff
@@ -0,0 +1,40 @@
+# Measurement Units
+
+TODO(hugovdm,younies): this is a work in progress. This document should be
+cleaned up before merging into the upstream unicode-org ICU repository. Some of
+this content belongs in header files.
+
+## Internal Implementation Details: ICU4C
```

Comment:
Implementation docs should go as comments in the header or source files.

Reply:
I'm moving this to a Google Doc - much easier to comment/discuss, and intended as a way to help us better structure our approach to ICU4J.